File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/model/message/incoming/PinsChangeMessage.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.model.message.incoming

import kotlinx.serialization.Serializable

@Serializable
data class PinsChangeMessage(
    override val operationId: String,
    val oldPin: String? = null,
    val newPin: String? = null
) : ChannelInMessage()

File: ios/Model/Message/Incoming/PinsChangeMessage.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import Foundation

class PinsChangeMessage: ChannelInMessage {
	// MARK: Properties

	let oldPin: String
	let newPin: String

	// MARK: Decodable

	enum CodingKeys: String, CodingKey {
		case oldPin
		case newPin
	}

	required init(from decoder: Decoder) throws {
		let container = try decoder.container(keyedBy: CodingKeys.self)
		self.oldPin = try container.decode(String.self, forKey: .oldPin)
		self.newPin = try container.decode(String.self, forKey: .newPin)
		try super.init(from: decoder)
	}
}
